crates/core/tedge_api/src/health.rs

```diff
@@ -1,12 +1,97 @@
 use std::process;
+use std::sync::Arc;
 
+use crate::mqtt_topics::ServiceTopicId;
 use mqtt_channel::Message;
 use mqtt_channel::PubChannel;
 use mqtt_channel::Topic;
 use mqtt_channel::TopicFilter;
 use serde_json::json;
 use time::OffsetDateTime;
 
+/// Encodes a valid health topic.
+///
+/// Health topics are topics on which messages about health status of services are published. To be
+/// able to send health messages, a health topic needs to be constructed for a given entity.
+// TODO: replace `Arc<str>` with `ServiceTopicId` after we're done with transition to new topics
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ServiceHealthTopic(Arc<str>);
+
+impl ServiceHealthTopic {
+    pub fn new(service: ServiceTopicId) -> Self {
+        // XXX: hardcoded MQTT root
+        ServiceHealthTopic(Arc::from(format!("te/{}/status/health", service.as_str())))
+    }
+
+    pub fn from_old_topic(topic: String) -> Result<Self, HealthTopicError> {
+        match topic.split('/').collect::<Vec<&str>>()[..] {
+            ["tedge", "health", _service_name] => {}
+            ["tedge", "health", _child_id, _service_name] => {}
+            _ => return Err(HealthTopicError),
+        }
+
+        Ok(Self(Arc::from(topic)))
+    }
+
+    pub fn is_health_topic(topic: &str) -> bool {
+        matches!(
+            topic.split('/').collect::<Vec<&str>>()[..],
+            ["te", _, _, _, _, "status", "health"]
+        )
+    }
+
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+
+    pub async fn send_health_status(&self, responses: &mut impl PubChannel) {
```

Comment:
I would really avoid to directly use a `PubChannel`.

Reply:
These functions are more or less direct translations of free functions below, just using the `ServiceHealthTopic` type. I'm not sure how to avoid using `PubChannel`, can you give and example of what you had in mind?

Reply:
Actually, `send_health_status` is not used and can be simply removed.